### PR TITLE
[DMS-77] Add usage examples to the documentation

### PR DESCRIPTION
**Pull request by s-and-witch**

### CODE REVIEW

**Comment by GoPavel** [comment]
I think it's better to add time and space complexity for `fromIter`, `put`, `delete` etc.
For some of them, it's obvious and for others, it's a repetition of what was written at the beginning of the doc page (in the overview), but I think it will be easier for a reader to have it repeated next to each function.

File: src/PersistentOrderedMap.mo [hunk 1]
```diff
@@ -102,7 +251,18 @@ module {
...
+  /// import Iter "mo:base/Iter"
+  ///
+  /// let mapOps = Map.MapOps<Nat>(Nat.compare);
+  /// let rbMap = mapOps.fromIter<Text>(Iter.fromArray([(0, "Zero"), (2, "Two"), (1, "One")]));
+  ///
+  /// Debug.print(debug_show(Iter.toArray(Map.iter(rbMap, #fwd))));
+  /// Debug.print(debug_show(Iter.toArray(Map.iter(rbMap, #bwd))));
+  ///
+  /// //  stdout [(0, "Zero"), (1, "One"), (2, "Two")]
+  /// //  stdout [(2, "Two"), (1, "One"), (0, "Zero")]
```
Comment (GoPavel) [inline src/PersistentOrderedMap.mo:265]:
```suggestion
  /// //  [(0, "Zero"), (1, "One"), (2, "Two")]
  /// //  [(2, "Two"), (1, "One"), (0, "Zero")]
```

**Comment by GoPavel** [comment]
Also a couple of high-level suggestions about documentation:
+ I think it would be good to mention the stability property somewhere in the overview (at the beginning)
+ It would be good to add some words about `MapOps` in its own doc section or in the overview.

File: src/PersistentOrderedMap.mo (hunk 1 above)
Comment (s-and-witch) [reply src/PersistentOrderedMap.mo]: Fixed

**Comment by s-and-witch** [comment]
@GoPavel updated patch according to your suggestions
- Added complexity information everywhere
- Mentioned, that it's stable (perhaps we may be even more explicit, but I don't know what to write there)
- Added a couple of words about `MapOps` and that it contains operations that either reshape or query `Map`.

**Pull request merged**